### ISSUE TITLE
Pod install fails after ejecting to ExpoKit

### ISSUE THREAD

**Issue by nickjanssen** [react-community/create-react-native-app#436]
## Expected Behavior

It seems CRNA's eject is broken:
```
create-react-native-app my-app
cd my-app
yarn eject
cd ios
pod install
```

## Observed Behavior
```bash
$ pod install
Analyzing dependencies
Fetching podspec for `DoubleConversion` from `../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec`
Pre-downloading: `ExpoKit` from `http://github.com/expo/expo.git`, tag `ios/1.20.5`
Fetching podspec for `Folly` from `../node_modules/react-native/third-party-podspecs/Folly.podspec`
Fetching podspec for `GLog` from `../node_modules/react-native/third-party-podspecs/GLog.podspec`
Fetching podspec for `React` from `../node_modules/react-native`
Fetching podspec for `Yoga` from `../node_modules/react-native/ReactCommon/yoga`
Downloading dependencies
Installing Amplitude-iOS (3.14.1)
Installing Analytics (3.6.7)
Installing AppAuth (0.90.0)
Installing Bolts (1.8.4)
Installing Branch (0.14.12)
Installing CocoaLumberjack (3.2.1)
Installing Crashlytics (3.9.0)
Installing DoubleConversion (1.1.5)
Installing ExpoKit (1.20.5)
Installing FBAudienceNetwork (4.26.0)
Installing FBSDKCoreKit (4.27.1)
Installing FBSDKLoginKit (4.27.1)
Installing FBSDKShareKit (4.27.1)
Installing Fabric (1.7.0)
Installing Folly (2016.09.26.00)
Installing GLog (0.3.4)
Installing GPUImage (0.1.7)
Installing Google-Mobile-Ads-SDK (7.22.0)
Installing GoogleAppUtilities (1.1.2)
Installing GoogleAuthUtilities (2.0.2)
Installing GoogleMaps (2.2.0)
Installing GoogleNetworkingUtilities (1.2.2)
Installing GoogleSignIn (3.0.0)
Installing GoogleSymbolUtilities (1.1.2)
Installing GoogleUtilities (1.3.2)
Installing KVOController (1.2.0)
Installing React (0.48.3)
Installing Yoga (0.48.3.React)
Installing boost (1.59.0)

[!] Error installing boost
[!] /usr/bin/tar xfz /var/folders/2p/zsx2_tb10hd74d7dyb3lvcp80000gn/T/d20171013-43428-ygkpqh/file.tgz -C /var/folders/2p/zsx2_tb10hd74d7dyb3lvcp80000gn/T/d20171013-43428-ygkpqh

tar: Unrecognized archive format
tar: Error exit delayed from previous errors.
```

## Environment

* `npm ls react-native-scripts`: (N/A since I ejected)
* `npm ls react-native`: 0.48.3
* `npm ls expo`: 21.0.2
* `node -v`: 6.9.5
* `npm -v`: 4.6.1
* `yarn --version`: 0.27.5
* `watchman version`: 4.6.0
* `pod --version`: 1.3.1

1. Operating system: macOS Sierra 10.12.4
2. Phone/emulator/simulator & version: N/A

**Comment by dikaiosune** [comment]
Hi! Can you try this again? That error looks like cocoapods failed to download the tarball.

**Comment by nickjanssen** [comment]
I tried it on a different laptop with a different account and the same error appears.

On Oct 18, 2017, at 4:11, Adam Perry <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi! Can you try this again? That error looks like cocoapods failed to download the tarball.

**Comment by brentvatne** [comment]
that's pretty strange -- boost is a popular package and we don't do anything different with it. perhaps there is a network problem? also hi @nickjanssen 👋

**Comment by sejr** [comment]
I am experiencing this issue. There is no network problem. @brentvatne

**Comment by dikaiosune** [comment]
Hi! I'm still not able to reproduce this issue, do you have any more information that would help us identify the cause here?

**Comment by IsaacWIvins** [comment]
Not sure if it's kosher but running `pod repo update` seemed to do the trick for me.

**Comment by nickjanssen** [comment]
`pod repo update` unfortunately didn't fix it for me. I still have the exact same error running `pod install` after that.

@dikaiosune I'll see if I can create a docker image to reproduce this.

@brentvatne Hey! ;)

**Comment by lakhman** [comment]
I had this same error, the spec download url points to a sourceforge page with a redirect, and that redirect is where it fails.

## [!] Error installing boost

>  126k    0     0  45312      0  0:00:02  0:00:02 --:--:--  101k
>    $ /usr/bin/tar xfz /var/folders/path/file.tgz -C /var/folders/path
>    tar: Unrecognized archive format
>    tar: Error exit delayed from previous errors.
> 
> [!] Error installing boost
> [!] /usr/bin/tar xfz /var/folders/path/file.tgz -C /var/folders/path


## Quick Fix

Find this file on your system and edit it's source url.

https://github.com/CocoaPods/Specs/blob/d0ec5a65e80656c8d78e12ff19f251df879e0bc2/Specs/9/9/d/boost/1.59.0/boost.podspec.json

### Find the boost spec file locally

```bash
open ~/.cocoapods/repos/master/Specs/9/9/d/boost/1.59.0/boost.podspec.json
``` 

### Change the source download url to a direct link

```javascript
  "source": {
    "http": "https://downloads.sourceforge.net/project/boost/boost/1.59.0/boost_1_59_0.tar.gz?r=&ts=1513105136&use_mirror=kent"
  },
```

### Pod install with patience

```bash
# everything should work - boost can be downloaded and unpacks
# Note this takes a while to compile, be patient.
pod install
```

**Comment by Epsiloni** [comment]
To build upon @lakhman's answer, what solved this for me was to go to the SourceForge page [here](https://downloads.sourceforge.net/project/boost/boost/1.59.0) and get the link it provides for you when you click the `boost_1_59_0.tar.gz` file (or simply [click here](https://downloads.sourceforge.net/project/boost/boost/1.59.0/boost_1_59_0.tar.gz)). You can get it by right clicking the "direct link" link they have and press "copy link".
Then paste it into the source like @lakhman showed above. and run `pod install`.

Be prepared, it takes forever.

**Comment by alcmoraes** [comment]
Somehow not even adding a direct link into podspec solved my problem, in my case what solved was to download the boost_1_59_0.tar.gz from sourceforge and serve it locally with a local http server, so afterwards I could edit the boost.podspec file pointing it to http://localhost:8080/boost_1_59_0.tar.gz

**Comment by anp** [comment]
Hi everyone! Looks like these issues should be cleared up by changes in https://github.com/facebook/react-native/pull/16756. Once that lands it'll make it out in a future release. If that doesn't fix these issues, please feel free to ping me here and I'll reopen or open a new issue. Thanks!

**Comment by corysimmons** [comment]
@anp still experiencing this

**Comment by anp** [comment]
@corysimmons CRNA doesn't support 0.54 yet because Expo had to delay our release: https://blog.expo.io/sdk-26-delayed-until-end-of-march-d02243e6ae7e

**Comment by corysimmons** [comment]
`pod repo update` seemed to do the trick for me. :\
